dvc/remote/http.py

```diff
@@ -36,7 +36,9 @@ def __init__(self, repo, config):
             )
 
     def _download(self, from_info, to_file, name=None, no_progress_bar=False):
-        request = self._request("GET", from_info.url, stream=True)
+        response = self._request("GET", from_info.url, stream=True)
+        if response.status_code != 200:
+            raise HTTPError(response.status_code, response.reason)
         with Tqdm(
             total=None if no_progress_bar else self._content_length(from_info),
```

Comment:
```suggestion
            total=None if no_progress_bar else self._content_length(response),
```
I wonder why we are still doing this extra request, also `._content_length()` doesn't need to work with url really.

Reply:
This is outside of your PR actually. But since you are here)

Reply:
But isn't the extra request done only if response does not contain `Content-Length`?

Reply:
If you pass url (path_info) and not response then you always make a second request. `_content_length()` should stop accepting urls and might be even inlined, this will prevent this inefficiency.

Reply:
That is right, sorry, I'll change accordingly. Though, I would leave `_content_length` as it is, in order to not obstruct `_download` method content.